hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/IncrementalInputSplits.java

```diff
@@ -447,8 +447,11 @@ private List<MergeOnReadInputSplit> getInputSplits(
                   .filter(logPath -> !logPath.endsWith(HoodieCDCUtils.CDC_LOGFILE_SUFFIX))
                   .collect(Collectors.toList()));
               String basePath = fileSlice.getBaseFile().map(BaseFile::getPath).orElse(null);
+              String sliceLastInstant = fileSlice.getLatestLogFile().isPresent()
+                      ? fileSlice.getLatestLogFile().get().getDeltaCommitTime()
+                      : fileSlice.getBaseInstantTime();
               return new MergeOnReadInputSplit(cnt.getAndAdd(1),
-                  basePath, logPaths, endInstant,
+                  basePath, logPaths, sliceLastInstant,
```

Comment:
Why the `endInstant` is incorrect?

Reply:
min(fileSlice.getLatestLogFile().map(HoodieInstant::getInstantTime).orElse(fileSlice.getBaseInstantTime()), endInstant),

You can move the method in `CompletionTimeQueryView.minInstant` to `HoodieTimeline` and reuse it here.

And also add a basic UT to check the end commit of the split.